modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/multiselect/MultiSelectBookmarksHelper.kt

```diff
@@ -0,0 +1,60 @@
+package au.com.shiftyjelly.pocketcasts.views.multiselect
+
+import android.content.res.Resources
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.map
+import au.com.shiftyjelly.pocketcasts.models.entity.Bookmark
+import au.com.shiftyjelly.pocketcasts.views.R
+import javax.inject.Inject
+import javax.inject.Singleton
+import au.com.shiftyjelly.pocketcasts.ui.R as UR
+
+@Singleton
+class MultiSelectBookmarksHelper @Inject constructor() : MultiSelectHelper<Bookmark>() {
+
+    override val toolbarActions: LiveData<List<MultiSelectAction>> = _selectedListLive
+        .map {
+            listOf(
+                MultiSelectAction.BookmarkAction.EditBookmark(isVisible = it.count() == 1),
+                MultiSelectAction.BookmarkAction.DeleteBookmark,
+                MultiSelectAction.SelectAll,
+            )
+        }
+
+    override lateinit var listener: Listener<Bookmark>
+
+    override fun isSelected(multiSelectable: Bookmark) =
+        selectedList.count { it.uuid == multiSelectable.uuid } > 0
+
+    override fun onMenuItemSelected(
+        itemId: Int,
+        resources: Resources,
+        fragmentManager: FragmentManager,
+    ): Boolean {
+        return when (itemId) {
+
+            UR.id.menu_edit -> {
+                // TODO: Bookmark - Add edit action
+                true
+            }
+
+            R.id.menu_delete -> {
+                // TODO: Bookmark - Add delete action
+                true
+            }
+
+            R.id.menu_select_all -> {
+                selectAll()
+                true
+            }
+
+            R.id.menu_unselect_all -> {
```

Comment:
I see there is an unselect all but if all options are selected this action isn't shown.

Reply:
Removed in c109ee0 for now. Might need it again for the deselect changes in https://github.com/Automattic/pocket-casts-android/pull/1172